rpc/src/server/testing.rs

```diff
@@ -4,57 +4,58 @@ use fnv::FnvHashSet;
 use futures::future::{AbortHandle, AbortRegistration};
 use futures::{Sink, Stream};
 use futures_test::task::noop_waker_ref;
-use pin_utils::{unsafe_pinned, unsafe_unpinned};
+use pin_project::pin_project;
 use std::collections::VecDeque;
 use std::io;
 use std::pin::Pin;
 use std::task::{Context, Poll};
 use std::time::SystemTime;
 
+#[pin_project]
 pub(crate) struct FakeChannel<In, Out> {
+    #[pin]
     pub stream: VecDeque<In>,
+    #[pin]
```

Comment:
Do these `VecDeque`s actually need to be pinned or do they impl unpin?

Reply:
That would require the inner type to be `Unpin`.